### ISSUE TITLE
[BUG] All links in the readme of @html-eslint/eslint-plugin expired, @html-eslint/eslint-plugin-react lost type prompts like 'recommended' or similar in htmlReact.configs

### ISSUE THREAD

**Issue by luo2430** [yeonjuan/html-eslint#628]

**Comment by yeonjuan** [comment]
@luo2430 Thank you for bringing this to our attention! This needs to be fixed.

**Comment by luo2430** [comment]
@yeonjuan 

> The documentation issue has indeed been fixed.

At runtime, `@html-eslint/eslint-plugin-react` correctly exports `configs.recommended` and `configs.all`:

```js
// lib/index.js
const plugin = {
  // ...
  configs: {
    recommended: { rules: allRules, plugins: { "@html-eslint/react": ... } },
    all:       { rules: allRules, plugins: { "@html-eslint/react": ... } },
  },
};
module.exports = plugin;
```

However, the type definition declares the entire plugin as a generic `ESLint.Plugin`:

```ts
// types/index.d.ts
declare const plugin: ESLint.Plugin;
export = plugin;
```

This causes `configs` to lose all concrete key information. As a result, when importing the plugin in a TypeScript project:

```ts
import htmlReact from "@html-eslint/eslint-plugin-react";

htmlReact.configs.recommended; // ❌ No autocomplete — configs is just `Record<string, ...>`
```

Neither `recommended` nor `all` shows up in IntelliSense, and the actual structure of each config entry is also untyped.

**Expected behavior:** The type definition should reflect the actual exports, similar to what `@html-eslint/eslint-plugin` already does in its `types/index.d.ts`.
